### PR TITLE
DOC: update 0.19.0 release notes

### PR DESCRIPTION
**Pull request by ev-br**
Comments are most appreciated since I might have missed things of note.

@pv @person142 could you have a look if this does justice to `LowLevelCallable` and `cython_special`, the two coolest things of this release?

### CODE REVIEW

File: doc/release/0.19.0-notes.rst [hunk 1]
```diff
@@ -119,9 +134,18 @@ memory usage.
...
+
+
 `scipy.special` improvements
 ----------------------------
 
+Scalar, typed versions of universal functions from `scipy.special` are available
+in the Cython space via ``cimport`` from the new module
+`scipy.special.cython_special`. These scalar functions can be expected to be
+significantly faster then the universal functions for scalar arguments.
+
```
Comment (person142) [inline doc/release/0.19.0-notes.rst]: Seems good to me. Maybe add a link to the section in the tutorial?

File: doc/release/0.19.0-notes.rst [hunk 2]
```diff
@@ -12,16 +12,31 @@ better documentation.  There have been a number of deprecations and
...
-will now shift to bug-fix releases on the 0.20.x branch, and on adding
+will now shift to bug-fix releases on the 0.19.x branch, and on adding
 new features on the master branch.
 
-This release requires Python 2.7 or 3.4-3.5 and NumPy 1.8.2 or greater.
+This release requires Python 2.7 or 3.4-3.6 and NumPy 1.8.2 or greater.
 
+Highlights of this release include:
+
+- A unified FFI layer, `scipy.LowLevelCallable`.
```
Comment (rgommers) [inline doc/release/0.19.0-notes.rst]: "FFI" --> "foreign function interface" (many readers won't know this acronym)

**Comment by rgommers** [comment]
Some new features worth mentioning:

- ``stats.multinomial``, ``stats.rv_histogram`` and ``stats.argus`` distributions were added
- support for ``float16`` inputs in ``fftpack``
- faster ``cluster.hierarchy.linkage`` for centroid and median methods (gh-6806)
- using weighted data in ``spatial.cKDTree.count_neighbors`` (gh-5647)
- ``signal.stft`` and ``signal.istft`` were added (gh-6058)

**Comment by e-q** [comment]
FYI: I'm adding `stft` to the release notes in gh-7036

**Comment by ev-br** [comment]
@e-q thanks! I'll cherry-pick 92f749658afe299164325034ef249c5b30f728ab if gh-7036 does not make it. Or you could just push it to this PR if you prefer.

File: doc/release/0.19.0-notes.rst (hunk 1 above)
Comment (ev-br) [inline doc/release/0.19.0-notes.rst]: https://github.com/scipy/scipy/pull/7035/commits/07721efc32e81cf83672501bf40bcae86481efeb refers to the tutorial. Not sure how to hyperlink to it however.

**Comment by ev-br** [comment]
Review comments addressed.

File: doc/release/0.19.0-notes.rst [hunk 3]
```diff
@@ -48,15 +63,20 @@ Moreover, both gained a ``balanced`` keyword to turn balancing on and off.
...
-`SphericalVoronoi` can handle > 200 k points (at least 10 million) & has
-improved performance
+`scipy.spatial.SphericalVoronoi` can handle > 200 k points (at least 10 million)
+and has improved performance.
 
 The function `scipy.spatial.distance.directed_hausdorff` was
 added to calculate the directed Hausdorff distance.
 
+``count_neighbors`` method of `scipy.spatial.cKDTree` gained an ability to
+perform weighted pair counting via the new keywords ``wights`` and
```
Comment (rgommers) [inline doc/release/0.19.0-notes.rst]: typo: ``wights`` --> ``weights``

File: doc/release/0.19.0-notes.rst [hunk 4]
```diff
@@ -106,6 +126,19 @@ substantially faster when operating on some n-dimensional arrays when n > 1.
...
+----------------------------
+
+Fast Fourier transform routines now accept `np.float16` inputs and upcast
+them to `np.float32`. Previously, they would raise an error.
+
+`scipy.cluster` improvements
+----------------------------
+
+Methods ``"centroid"`` and ``"median"`` of `scipy.cluster.hierarchy.linkage`
+have been significantly sped up.
```
Comment (rgommers) [inline doc/release/0.19.0-notes.rst]: Maybe another sentence to add: 

    Long-standing issues with using ``linkage`` on large input data (over 16 GB) have been resolved.

**Review by larsoner — COMMENTED** [review]
Changes LGTM (once @rgommers's comments are addressed)

**Comment by ev-br** [comment]
Comments addressed (Thanks Ralf)

File: doc/release/0.19.0-notes.rst [hunk 5]
```diff
@@ -93,9 +113,9 @@ response from second-order sections.
 The function `scipy.signal.unit_impulse` was added to conveniently
 generate an impulse function.
 
-The function `scipy.signal.irrnotch` was added to design second-order
+The function `scipy.signal.iirnotch` was added to design second-order
 IIR notch filters that can be used to remove a frequency component from
-a signal. The dual function  `scipy.signal.irrpeak` was added to
+a signal. The dual function  `scipy.signal.iirpeak` was added to
 compute the coeficients of a second-order IIR peak (resonant) filter.
```
Comment (thisch) [inline doc/release/0.19.0-notes.rst]: typo: coeff instead of coef

Comment (ev-br) [inline doc/release/0.19.0-notes.rst]: Good catch, thanks!

**Comment by ev-br** [comment]
Merging. Thanks all for the reviews.

**Pull request merged**